File: src/main/java/no/novari/flyt/integration/validation/IntegrationValidationContext.java
package no.novari.flyt.integration.validation;

import jakarta.validation.Payload;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.extern.jackson.Jacksonized;
import no.novari.flyt.integration.model.dtos.ConfigurationDto;

@Getter
@EqualsAndHashCode
@Jacksonized
@Builder
public class IntegrationValidationContext implements Payload {
    private final Long integrationId;
    private final ConfigurationDto configuration;
}

File: src/main/kotlin/no/novari/flyt/integration/validation/IntegrationValidationContext.kt
package no.novari.flyt.integration.validation

import jakarta.validation.Payload
import no.novari.flyt.integration.api.dto.ConfigurationDto

data class IntegrationValidationContext(
    val integrationId: Long? = null,
    val configuration: ConfigurationDto? = null,
) : Payload
